src/fairreckitlib/model/algorithms/elliot/elliot_recommender.py

```diff
@@ -32,17 +32,17 @@ def __init__(self, name: str, params: Dict[str, Any], **kwargs):
         if not self.rated_items_filter:
             raise RuntimeError('no rated items filter is not supported.')
 
-    def train(self, train_set: pd.DataFrame) -> None:
+    def on_train(self) -> None:
         """Train the elliot model not supported."""
         # not used, training is done by running the framework
         raise NotImplementedError()
```

Comment:
If a method is not supposed to be implemented, it shouldn't raise a NotImplementedError.
https://docs.python.org/3/library/exceptions.html#NotImplementedError

Reply:
good point, ill change them to RuntimeError's as well